charts/redpanda/templates/statefulset.yaml

```diff
@@ -237,13 +237,24 @@ spec:
               valueFrom:
                 fieldRef:
                   fieldPath: status.hostIP
+          # finish the lifecycle scripts with "true" to prevent them from terminating the pod prematurely
           lifecycle:
             postStart:
               exec:
-                command: ["/bin/bash", "-c", "./var/lifecycle/postStart.sh"]
+                command:
+                  - /bin/bash
+                  - -c
+                  - |
+                    timeout {{ div .Values.statefulset.terminationGracePeriodSeconds 2 }} /var/lifecycle/postStart.sh
+                    true
```

Comment:
The pod's already started. If it fails we don't want to terminate the pod.

Reply:
Interesting if it fails why we use postStart hook at all?

Reply:
Yeah. It would be nice if we could predict when it would fail and either fix what's causing the failure or not even try, but I don't know any way to do that preemptively.

Reply:
The reason we run this at all is to bring the broker out of maintenance mode (and something to do with a user). Tests have shown that this is effective and does what we intend. If we find that there are failures and can find a cause that we can safely correct, we can add that code at that time.